frontend/src/views/admin/matching/index.tsx

```diff
@@ -21,11 +160,261 @@ export function AdminMatchingView() {
+
+                // Mark positions as being applied for
+                for (const positionPreference of combinedApplication.position_preferences) {
+                    const savedMatchForPosition =
+                        savedMatches?.find(
+                            (match) =>
+                                match.positionCode ===
+                                    positionPreference.position.position_code &&
+                                match.utorid === applicant.utorid
+                        ) || null;
+
+                    initialMatches.push({
+                        utorid: applicant.utorid,
+                        positionCode: positionPreference.position.position_code,
+                        status: savedMatchForPosition
+                            ? savedMatchForPosition.status
+                            : "applied",
+                        hoursAssigned: savedMatchForPosition
+                            ? savedMatchForPosition.hoursAssigned
+                            : 0,
+                    });
+                }
+            }
+
+            // Mark positions as being assigned
+            for (const assignment of assignments) {
+                const matchingAssignment = initialMatches.find(
+                    (match) =>
+                        match.utorid === assignment.applicant.utorid &&
+                        match.positionCode === assignment.position.position_code
+                );
+
+                // Update existing match object if it exists
+                if (matchingAssignment) {
+                    matchingAssignment.status = "assigned";
+                    matchingAssignment.hoursAssigned = assignment.hours
+                        ? assignment.hours
+                        : 0;
+                } else {
+                    // Otherwise, create a new one
+                    initialMatches.push({
+                        utorid: assignment.applicant.utorid,
+                        positionCode: assignment.position.position_code,
+                        status: "assigned",
+                        hoursAssigned: assignment.hours ? assignment.hours : 0,
+                    });
+                }
+            }
+
+            return await dispatch(batchUpsertMatches(initialMatches));
+        }
+
+        async function initializeGuarantees() {
+            const savedGuarantees: AppointmentGuaranteeStatus[] =
+                localStore.guarantees;
+            if (savedGuarantees.length > 0) {
+                return await dispatch(batchUpsertGuarantees(savedGuarantees));
+            }
+        }
+
+        async function initializeNotes() {
+            const savedNotes: Record<string, string | null> = localStore.notes;
+            if (Object.keys(savedNotes).length > 0) {
+                return await dispatch(batchUpsertNotes(savedNotes));
+            }
+        }
+
+        initializeMatches();
+        initializeGuarantees();
+        initializeNotes();
```

Comment:
I'm confused why all these are async. There shouldn't be anything that needs async with staging data/actions.